platform/backend_connectors/hydrolix_backend_connector.go

```diff
@@ -48,5 +55,34 @@ func (p *HydrolixBackendConnector) InstanceName() string {
 }
 
 func (p *HydrolixBackendConnector) Exec(ctx context.Context, query string, args ...interface{}) error {
+	if p.IngestURL == "" || p.AccessToken == "" {
+		logger.Info().Msg("missing ingest URL or access token")
+		// TODO for fallback, execute the query directly on the database connection
+		_, err := p.connection.ExecContext(ctx, query)
+		return err
+	}
+
+	// Create HTTP request using the JSON payload from query
```

Comment:
Suggestion: since this `Exec` is on a more permissive side I'd (at least for the time being of this PoC) put a little more defensive measure -> marshal and check if we're dealing with valid JSON here.
That'd catch connector misconfiguration resulting in sending raw SQL to Hydrolix.